File: cpp_src/LongestCommonPrefix.cpp
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        int size = strs.size();
        if(size == 0){
            return "";
        }
        sort(strs.begin(),strs.end());
        int length = strs[0].size();
        for(int i = 0; i < length; i++){
            if(strs[0][i] != strs[size - 1][i]){
                return strs[0].substr(0,i);
            }
        }
        return strs[0];
        
    }
};

File: python_src/LongestCommonPrefix.py
class Solution(object):
    def longestCommonPrefix(self, strs):
        """
        :type strs: List[str]
        :rtype: str
        """
        if len(strs) == 0:
            return ""
        for i in range(len(strs[0])):
            if all([ i < len(s) and s[i] == strs[0][i] for s in strs]):
                continue
            else:
                return strs[0][0:i]
        else:
            return strs[0]
        
        